airbyte-ci/connectors/connector_ops/connector_ops/required_reviewer_checks.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Dict, List, Set, Union
+
+import yaml
+from connector_ops import utils
+
+BACKWARD_COMPATIBILITY_REVIEWERS = {"connector-operations", "connector-extensibility"}
+TEST_STRICTNESS_LEVEL_REVIEWERS = {"connector-operations"}
+GA_BYPASS_REASON_REVIEWERS = {"connector-operations"}
+GA_CONNECTOR_REVIEWERS = {"gl-python"}
+BREAKING_CHANGE_REVIEWERS = {"breaking-change-reviewers"}
+REVIEW_REQUIREMENTS_FILE_PATH = ".github/connector_org_review_requirements.yaml"
+
+
+def find_changed_important_connectors() -> Set[utils.Connector]:
+    """Find important connectors modified on the current branch.
+
+    Returns:
+        Set[utils.Connector]: The set of GA connectors that were modified on the current branch.
+    """
+    changed_connectors = utils.get_changed_connectors(destination=False, third_party=False)
+    return {connector for connector in changed_connectors if connector.is_important_connector}
+
+
+def get_bypass_reason_changes() -> Set[utils.Connector]:
+    """Find connectors that have modified bypass_reasons.
+
+    Returns:
+        Set[str]: Set of connector names e.g {"source-github"}: The set of GA connectors that have changed bypass_reasons.
+    """
+    bypass_reason_changes = utils.get_changed_acceptance_test_config(diff_regex="bypass_reason")
+    return bypass_reason_changes.intersection(find_changed_important_connectors())
+
+
+def find_mandatory_reviewers() -> List[Dict[str, Union[str, Dict[str, List]]]]:
+    important_connector_changes = find_changed_important_connectors()
+    backward_compatibility_changes = utils.get_changed_acceptance_test_config(diff_regex="disable_for_version")
+    test_strictness_level_changes = utils.get_changed_acceptance_test_config(diff_regex="test_strictness_level")
+    ga_bypass_reason_changes = get_bypass_reason_changes()
+    breaking_change_changes = utils.get_changed_metadata(diff_regex="upgradeDeadline")
+
+    required_reviewers = []
+
+    if backward_compatibility_changes:
+        required_reviewers.append({"name": "Backwards Compatibility Test Skip", "teams": list(BACKWARD_COMPATIBILITY_REVIEWERS)})
+    if test_strictness_level_changes:
+        required_reviewers.append({"name": "Acceptance Test Strictness Level", "teams": list(TEST_STRICTNESS_LEVEL_REVIEWERS)})
+    if ga_bypass_reason_changes:
+        required_reviewers.append({"name": "GA Acceptance Test Bypass", "teams": list(GA_BYPASS_REASON_REVIEWERS)})
+    if important_connector_changes:
+        required_reviewers.append({"name": "GA Connectors", "teams": list(GA_CONNECTOR_REVIEWERS)})
+    if breaking_change_changes:
+        required_reviewers.append({"name": "Breaking Changes", "teams": list(BREAKING_CHANGE_REVIEWERS)})
```

Comment:
a mapping between change detection function and mandatory reviewers can be more readable